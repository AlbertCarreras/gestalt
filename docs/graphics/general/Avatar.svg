<svg width="325" height="180" viewBox="0 0 325 180" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect width="325" height="180" fill="#F6FDF5"/>
<circle cx="162" cy="90" r="35" fill="#005C62"/>
<path d="M146 90C146 81.1625 153.163 74 162 74C170.837 74 178 81.1625 178 90C178 98.8375 170.837 106 162 106C153.163 106 146 98.8375 146 90ZM156.256 94.3438C155.894 93.925 155.262 93.8812 154.844 94.2437C154.425 94.6062 154.381 95.2375 154.744 95.6562C156.131 97.2563 158.556 99 161.944 99C165.444 99 167.863 97.2563 169.256 95.6562C169.619 95.2375 169.575 94.6062 169.156 94.2437C168.738 93.8812 168.106 93.925 167.744 94.3438C166.619 95.6375 164.713 97 161.944 97C159.288 97 157.375 95.6375 156.256 94.3438ZM157.025 85C155.919 85 155.025 85.8937 155.025 87C155.025 88.1063 155.919 89 157.025 89C158.125 89 159.025 88.1063 159.025 87C159.025 85.8937 158.125 85 157.025 85ZM167.025 89C168.125 89 169.025 88.1063 169.025 87C169.025 85.8937 168.125 85 167.025 85C165.919 85 165.025 85.8937 165.025 87C165.025 88.1063 165.919 89 167.025 89Z" fill="#CDF9CD"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M162 123C180.225 123 195 108.225 195 90C195 71.7746 180.225 57 162 57C143.775 57 129 71.7746 129 90C129 108.225 143.775 123 162 123ZM162 125C181.33 125 197 109.33 197 90C197 70.67 181.33 55 162 55C142.67 55 127 70.67 127 90C127 109.33 142.67 125 162 125Z" fill="white"/>
<path d="M0 0V-2H-2V0H0ZM325 0H327V-2H325V0ZM325 180V182H327V180H325ZM0 180H-2V182H0V180ZM0 2H325V-2H0V2ZM323 0V180H327V0H323ZM325 178H0V182H325V178ZM2 180V0H-2V180H2Z" fill="#767676"/>
</svg>
