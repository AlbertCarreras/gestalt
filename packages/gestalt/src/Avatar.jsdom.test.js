// @flow strict
import { fireEvent, render } from '@testing-library/react';
import Avatar from './Avatar.js';

describe('Avatar', () => {
  it('renders multi-byte character initial', () => {
    const { container } = render(<Avatar name="ðŸ’© astral" />);
    expect(container).toMatchSnapshot();
  });

  it('renders with an empty name shows default icon', () => {
    const { container } = render(<Avatar name="" />);
    expect(container).toMatchSnapshot();
  });

  it('renders an outline', () => {
    const { container } = render(<Avatar name="Jenny" outline />);
    expect(container).toMatchSnapshot();
  });

  it('renders the verified icon', () => {
    const { container } = render(<Avatar name="Jammy" verified />);
    expect(container).toMatchSnapshot();
  });

  it('renders the correct src', () => {
    const { container } = render(
      <Avatar name="Strava" src="http://pinterest.com/img/strave.png" />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the correct size - xs', () => {
    const { container } = render(
      <Avatar name="Strava" src="http://pinterest.com/img/strave.png" size="xs" />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the correct size - sm', () => {
    const { container } = render(
      <Avatar name="Strava" src="http://pinterest.com/img/strave.png" size="sm" />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the correct size - md', () => {
    const { container } = render(
      <Avatar name="Strava" src="http://pinterest.com/img/strave.png" size="md" />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the correct size - lg', () => {
    const { container } = render(
      <Avatar name="Strava" src="http://pinterest.com/img/strave.png" size="lg" />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the correct size - xl', () => {
    const { container } = render(
      <Avatar name="Strava" src="http://pinterest.com/img/strave.png" size="xs" />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the checkmark on verified default', () => {
    const { container } = render(
      <Avatar name="Strava" src="http://pinterest.com/img/strave.png" size="md" verified />,
    );
    expect(container).toMatchSnapshot();
  });

  it('Avatar handles Image error by rendering the default avatar', () => {
    const { getByAltText, getByText } = render(<Avatar name="Name" src="example.com" />);
    // eslint-disable-next-line testing-library/prefer-screen-queries -- Please fix the next time this file is touched!
    fireEvent.error(getByAltText('Name'));

    // eslint-disable-next-line testing-library/prefer-screen-queries -- Please fix the next time this file is touched!
    expect(getByText('N')).toBeTruthy();
    expect(() => {
      // eslint-disable-next-line testing-library/prefer-screen-queries -- Please fix the next time this file is touched!
      getByText('T');
    }).toThrow('Unable to find an element with the text: T');
  });
});
