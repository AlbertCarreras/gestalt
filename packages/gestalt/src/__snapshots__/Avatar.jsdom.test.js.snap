// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders an outline 1`] = `
<div>
  <div
    class="box circle relative"
    data-test-id="gestalt-avatar-svg"
    style="box-shadow: 0 0 0 1px #fff; width: 100%;"
  >
    <div
      class="box circle relative secondary"
      style="padding-bottom: 100%;"
    >
      <div
        class="absolute bottom0 box justifyCenter left0 right0 top0 xsDisplayFlex"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          version="1.1"
          viewBox="-50 -50 100 100"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Jenny
          </title>
          <text
            class="antialiased sansSerif fontWeightSemiBold "
            dy="0.35em"
            fill="#111"
            font-size="40px"
            text-anchor="middle"
          >
            J
          </text>
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Avatar renders multi-byte character initial 1`] = `
<div>
  <div
    class="box circle relative"
    data-test-id="gestalt-avatar-svg"
    style="width: 100%;"
  >
    <div
      class="box circle relative secondary"
      style="padding-bottom: 100%;"
    >
      <div
        class="absolute bottom0 box justifyCenter left0 right0 top0 xsDisplayFlex"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          version="1.1"
          viewBox="-50 -50 100 100"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            ðŸ’© astral
          </title>
          <text
            class="antialiased sansSerif fontWeightSemiBold "
            dy="0.35em"
            fill="#111"
            font-size="40px"
            text-anchor="middle"
          >
            ðŸ’©
          </text>
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Avatar renders the checkmark on verified default 1`] = `
<div>
  <div
    class="box circle relative"
    data-test-id="gestalt-avatar-svg"
    style="width: 48px; height: 48px;"
  >
    <div
      class="Mask circle willChangeTransform"
    >
      <div
        class="box relative"
        style="background-color: rgb(239, 239, 239); padding-bottom: 100%;"
      >
        <img
          alt="Strava"
          class="img"
          fetchpriority="auto"
          loading="auto"
          src="http://pinterest.com/img/strave.png"
        />
      </div>
      <div
        class="wash"
      />
    </div>
    <div
      class="absolute box"
      style="width: 25%; height: 25%; min-width: 12px; min-height: 12px; bottom: 4%; right: 4%;"
    >
      <div
        class="box circle default"
        style="width: 100%; height: 100%;"
      >
        <svg
          aria-hidden="true"
          aria-label=""
          class="icon brandPrimaryIcon iconBlock"
          height="100%"
          role="img"
          viewBox="0 0 24 24"
          width="100%"
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Avatar renders the correct size - lg 1`] = `
<div>
  <div
    class="box circle relative"
    data-test-id="gestalt-avatar-svg"
    style="width: 64px; height: 64px;"
  >
    <div
      class="Mask circle willChangeTransform"
    >
      <div
        class="box relative"
        style="background-color: rgb(239, 239, 239); padding-bottom: 100%;"
      >
        <img
          alt="Strava"
          class="img"
          fetchpriority="auto"
          loading="auto"
          src="http://pinterest.com/img/strave.png"
        />
      </div>
      <div
        class="wash"
      />
    </div>
  </div>
</div>
`;

exports[`Avatar renders the correct size - md 1`] = `
<div>
  <div
    class="box circle relative"
    data-test-id="gestalt-avatar-svg"
    style="width: 48px; height: 48px;"
  >
    <div
      class="Mask circle willChangeTransform"
    >
      <div
        class="box relative"
        style="background-color: rgb(239, 239, 239); padding-bottom: 100%;"
      >
        <img
          alt="Strava"
          class="img"
          fetchpriority="auto"
          loading="auto"
          src="http://pinterest.com/img/strave.png"
        />
      </div>
      <div
        class="wash"
      />
    </div>
  </div>
</div>
`;

exports[`Avatar renders the correct size - sm 1`] = `
<div>
  <div
    class="box circle relative"
    data-test-id="gestalt-avatar-svg"
    style="width: 32px; height: 32px;"
  >
    <div
      class="Mask circle willChangeTransform"
    >
      <div
        class="box relative"
        style="background-color: rgb(239, 239, 239); padding-bottom: 100%;"
      >
        <img
          alt="Strava"
          class="img"
          fetchpriority="auto"
          loading="auto"
          src="http://pinterest.com/img/strave.png"
        />
      </div>
      <div
        class="wash"
      />
    </div>
  </div>
</div>
`;

exports[`Avatar renders the correct size - xl 1`] = `
<div>
  <div
    class="box circle relative"
    data-test-id="gestalt-avatar-svg"
    style="width: 24px; height: 24px;"
  >
    <div
      class="Mask circle willChangeTransform"
    >
      <div
        class="box relative"
        style="background-color: rgb(239, 239, 239); padding-bottom: 100%;"
      >
        <img
          alt="Strava"
          class="img"
          fetchpriority="auto"
          loading="auto"
          src="http://pinterest.com/img/strave.png"
        />
      </div>
      <div
        class="wash"
      />
    </div>
  </div>
</div>
`;

exports[`Avatar renders the correct size - xs 1`] = `
<div>
  <div
    class="box circle relative"
    data-test-id="gestalt-avatar-svg"
    style="width: 24px; height: 24px;"
  >
    <div
      class="Mask circle willChangeTransform"
    >
      <div
        class="box relative"
        style="background-color: rgb(239, 239, 239); padding-bottom: 100%;"
      >
        <img
          alt="Strava"
          class="img"
          fetchpriority="auto"
          loading="auto"
          src="http://pinterest.com/img/strave.png"
        />
      </div>
      <div
        class="wash"
      />
    </div>
  </div>
</div>
`;

exports[`Avatar renders the correct src 1`] = `
<div>
  <div
    class="box circle relative"
    data-test-id="gestalt-avatar-svg"
    style="width: 100%;"
  >
    <div
      class="Mask circle willChangeTransform"
    >
      <div
        class="box relative"
        style="background-color: rgb(239, 239, 239); padding-bottom: 100%;"
      >
        <img
          alt="Strava"
          class="img"
          fetchpriority="auto"
          loading="auto"
          src="http://pinterest.com/img/strave.png"
        />
      </div>
      <div
        class="wash"
      />
    </div>
  </div>
</div>
`;

exports[`Avatar renders the verified icon 1`] = `
<div>
  <div
    class="box circle relative"
    data-test-id="gestalt-avatar-svg"
    style="width: 100%;"
  >
    <div
      class="box circle relative secondary"
      style="padding-bottom: 100%;"
    >
      <div
        class="absolute bottom0 box justifyCenter left0 right0 top0 xsDisplayFlex"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          version="1.1"
          viewBox="-50 -50 100 100"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Jammy
          </title>
          <text
            class="antialiased sansSerif fontWeightSemiBold "
            dy="0.35em"
            fill="#111"
            font-size="40px"
            text-anchor="middle"
          >
            J
          </text>
        </svg>
      </div>
    </div>
    <div
      class="absolute box"
      style="width: 25%; height: 25%; min-width: 12px; min-height: 12px; bottom: 4%; right: 4%;"
    >
      <div
        class="box circle default"
        style="width: 100%; height: 100%;"
      >
        <svg
          aria-hidden="true"
          aria-label=""
          class="icon brandPrimaryIcon iconBlock"
          height="100%"
          role="img"
          viewBox="0 0 24 24"
          width="100%"
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Avatar renders with an empty name shows default icon 1`] = `
<div>
  <div
    class="box circle relative"
    data-test-id="gestalt-avatar-svg"
    style="width: 100%;"
  >
    <div
      class="box circle relative secondary"
      style="padding-bottom: 100%;"
    >
      <div
        class="absolute bottom0 box justifyCenter left0 right0 top0 xsDisplayFlex"
      >
        <svg
          preserveAspectRatio="xMidYMid meet"
          version="1.1"
          viewBox="-50 -50 100 100"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <text
            class="antialiased sansSerif fontWeightSemiBold "
            dy="0.35em"
            fill="#111"
            font-size="40px"
            text-anchor="middle"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
